pre_commit/runner.py

```diff
@@ -16,26 +16,27 @@ class Runner(object):
     repository under test.
     """
 
-    def __init__(self, git_root):
+    def __init__(self, git_root, config_file=None):
         self.git_root = git_root
+        self.config_file = config_file or C.CONFIG_FILE
 
     @classmethod
-    def create(cls):
+    def create(cls, config_file=None):
```

Comment:
I'd make this a required argument (in both places), I know this is going to require a bunch of test changes but it'll make the runtime much simpler and less error-prone probably?